### PR TITLE
fix FilterTypes typing @types/react-table

### PR DESCRIPTION
**Pull request by akomm**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/55920

### CODE REVIEW

**Comment by akomm** [comment]
Test was already there, but not type-checked because the typing was "any" before

**Review by ggascoigne — COMMENTED** [review]
I saw your other post about not using this lib, so I appreciate you taking the time to post this.  That said, if you'd like me to take it over and fix the tests I completely understand.  I's post a fix, but I've got to remember how to best post to someone else's PR.

File: types/react-table/index.d.ts [hunk 1]
```diff
@@ -474,7 +474,8 @@ export interface UseFiltersColumnProps<D extends object> {
 export type FilterProps<D extends object> = HeaderProps<D>;
 export type FilterValue = any;
 export type Filters<D extends object> = Array<{ id: IdType<D>; value: FilterValue }>;
-export type FilterTypes<D extends object> = Record<string, FilterValue>;
+export type FilterTypes<D extends object> = Record<string, FilterType<D>>;
+export type FilterType<D extends object> = (rows: Array<Row<D>>, columnIds: IdType<D>, filterValue: FilterValue) => Array<Row<D>>;
```
[the remark this answers is not in the document]
Comment (akomm) [reply types/react-table/index.d.ts]: I based my test correct assumption on the example provided by react-table page, which are as it seems also wrong:
https://codesandbox.io/s/github/tannerlinsley/react-table/tree/master/examples/filtering?file=/src/App.js:5440-5771

I assume the test here was based on it too?

Looks like the test is not actually run - just linted. And the `test-all` - if that runs any tests and not just lints all packages - fails due to some unrelated issue after running a while. Going to force push here and let the CI run it...

**Review by ggascoigne — APPROVED** [review]
Thank you

**Comment by akomm** [comment]
Ready to merge

**Pull request merged**